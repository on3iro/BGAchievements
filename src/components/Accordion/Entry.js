/**
 * A single accordion entry. Should receive a handleExpand function as props
 * which in turn is called whenever the user clicks on the entry question.
  */

import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

export const Wrapper = styled.div`
  margin-bottom: 20px;
  border: 1px solid #e3e3e3;
`

export const Question = styled.p`
  background-color: #e3e3e3;
  margin: 0;
  padding: 10px;
  cursor: pointer;
`

export const Answer = styled.div`
  margin: 0;
  padding: 10px;
`

const Entry = ({ handleExpand, expanded = false, question, answer }) => {
  return (
    <Wrapper>
      <Question onClick={handleExpand}>{question}</Question>
      {
        expanded &&
        <Answer>{answer()}</Answer>
      }
    </Wrapper>
  )
}

Entry.propTypes = {
  handleExpand: PropTypes.func.isRequired,
  question: PropTypes.string.isRequired,
  answer: PropTypes.func.isRequired,
  expanded: PropTypes.bool
}

export default Entry
