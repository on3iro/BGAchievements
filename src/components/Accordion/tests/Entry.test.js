import React from 'react'
import { shallow, mount } from 'enzyme'

import Entry from '../Entry'
import {
  Wrapper,
  Question,
  Answer
} from '../Entry'

describe('<Entry />', () => {
  it('should render Wrapper', () => {
    const wrapper = shallow(
      <Entry
        handleExpand={() => {}}
        question='foo'
        answer={() => 'bar'}
      />
    )

    expect(wrapper.find(Wrapper).length).toBe(1)
  })

  it('should render Question', () => {
    const wrapper = shallow(
      <Entry
        handleExpand={() => {}}
        question='foo'
        answer={() => 'bar'}
      />
    )

    expect(wrapper.find(Question).length).toBe(1)
  })

  it('should render Answer', () => {
    const wrapper = shallow(
      <Entry
        handleExpand={() => {}}
        expanded
        question='foo'
        answer={() => 'bar'}
      />
    )

    expect(wrapper.find(Answer).length).toBe(1)
  })

  it('should render <div> when folded', () => {
    const wrapper = mount(
      <Entry
        handleExpand={() => {}}
        question='How are you?'
        answer={() => 'bar'}
      />
    )
    expect(wrapper.find('div').length).toMatchSnapshot()
  })

  it('should render answer <div> when expanded', () => {
    const wrapper = mount(
      <Entry
        handleExpand={() => {}}
        expanded
        question='How are you?'
        answer={() => 'fine'}
      />
    )
    expect(wrapper.find('div').length).toMatchSnapshot()
  })

  it('should expand on click on question', () => {
    const expandSpy = jest.fn()
    const wrapper = mount(
      <Entry
        handleExpand={expandSpy}
        question='How are you?'
        answer={() => 'bar'}
      />
    )

    wrapper.find(Question).first().simulate('click')
    expect(expandSpy).toHaveBeenCalled()
  })
})
