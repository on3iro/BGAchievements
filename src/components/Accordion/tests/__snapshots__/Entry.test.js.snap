exports[`<Entry /> should render <div> when folded 1`] = `1`

exports[`<Entry /> should render answer <div> when expanded 1`] = `2`
